rt_gene_model_training/train_blink_model.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+
+# from https://hackernoon.com/tf-serving-keras-mobilenetv2-632b8d92983c
+
+from dataset_manager import RT_BENE
+
+from blink_model_factory import create_model, load_existing_model
+from tensorflow.keras.callbacks import ModelCheckpoint
+
+from tqdm import tqdm
+
+import gc
+
+import argparse
+
+import numpy as np
+
+import time
+import json
+
+import tensorflow as tf
+
+config = tf.ConfigProto()
+config.gpu_options.allow_growth = True
+tf.keras.backend.set_session(tf.Session(config=config))
+
+# from https://stackoverflow.com/questions/12116685/how-can-i-require-my-python-scripts-argument-to-be-a-float-between-0-0-1-0-usin
+def restricted_float(x):
+    x = float(x)
+    if x < 0.0 or x > 1.0:
+        raise argparse.ArgumentTypeError("%r not in range [0.0, 1.0]"%(x,))
+    return x
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("model_base", choices=['densenet121', 'resnet50', 'mobilenetv2'])
+    parser.add_argument("model_path", help="target folder to save the models (auto-saved)")
+    parser.add_argument("csv_subjects", help="")
+    #parser.add_argument("dataset_imgs", help="")
+    parser.add_argument("--use_weights", help="whether to use weights")
+    parser.add_argument("--random_subset", type=restricted_float, help="")
+    parser.add_argument("--batch_size", type=int, help="", default=64)
+    parser.add_argument("--epochs", type=int, help="", default=8)
+    parser.add_argument("--input_size", type=int, help="", default=96)
+
+    args = parser.parse_args()
+    model_base = args.model_base
+    batch_size = args.batch_size
+    epochs = args.epochs
+
+    input_size = args.input_size
+
+    if args.use_weights:
+        weight_factor = 'use_weight'
+    else:
+        weight_factor = 'no_weight'
+
+    dataset = RT_BENE(args.csv_subjects, input_size, args.random_subset)
+    dataset.load()
+    fold_infos = [([0, 1], [2], 'fold1'), ([0, 2], [1], 'fold2'), ([1, 2], [0], 'fold3')]
+
+    # 3 folds training
+    for subjects_train, subjects_test, fold_name in fold_infos:
+        training_fold = dataset.get_folds(subjects_train)
+        validation_fold = dataset.get_folds(subjects_test)
```

Comment:
I think there is some issue here. There should be three sets of data: train, test and validation. Here, validation == test, which is not correct. I think this needs fixing. Let me know if you have questions regarding this.